tdp/core/models/deployment_log.py

```diff
@@ -74,41 +75,43 @@ class DeploymentLog(Base):
     Hold past and current deployment information.
 
     Attributes:
-        id (int): Deployment log id.
-        sources (List[str]): List of source nodes, in the case of Dag deployment type.
-        targets (List[str]): List of target nodes, in the case of Dag deployment type. List of operations, in the case of Run deployment type.
-        filter_expression (str): Filter expression.
-        filter_type (FilterTypeEnum): Filter type.
-        hosts (List[str]): List of hosts.
-        start_time (datetime): Deployment start time.
-        end_time (datetime): Deployment end time.
-        state (DeploymentStateEnum): Deployment state.
-        deployment_type (DeploymentTypeEnum): Deployment type.
-        restart (bool): Restart flag.
+        id: Deployment log id.
+        sources: List of source nodes, in the case of Dag deployment type.
+        targets: List of target nodes, in the case of Dag deployment type.
+          List of operations, in the case of Run deployment type.
+        filter_expression: Filter expression.
+        filter_type: Filter type.
+        hosts: List of hosts.
+        start_time: Deployment start time.
+        end_time: Deployment end time.
+        state: Deployment state.
+        deployment_type: Deployment type.
+        restart: Restart flag.
     """
 
     __tablename__ = "deployment_log"
 
-    id = Column(Integer, primary_key=True)
-    sources = Column(JSON(none_as_null=True))
-    targets = Column(JSON(none_as_null=True))
-    filter_expression = Column(String(length=OPERATION_NAME_MAX_LENGTH * 5))
-    filter_type = Column(Enum(FilterTypeEnum))
-    hosts = Column(JSON(none_as_null=True))
-    start_time = Column(DateTime(timezone=False))
-    end_time = Column(DateTime(timezone=False))
-    state: Column = Column(Enum(DeploymentStateEnum))
-    deployment_type: Column = Column(Enum(DeploymentTypeEnum))
-    restart = Column(Boolean, default=False)
-
-    operations = relationship(
-        "OperationLog",
+    id: Mapped[int] = mapped_column(primary_key=True)
+    sources: Mapped[Optional[list]] = mapped_column(JSON(none_as_null=True))
```

Comment:
You can add the type for item in the list or it does not work ?
```suggestion
    sources: Mapped[Optional[list[str]]] = mapped_column(JSON(none_as_null=True))
```

Reply:
Yes I can!

Reply:
Done